internal/handlers/common/update.go

```diff
@@ -918,23 +918,56 @@ func validateOperatorKeys(command string, docs ...*types.Document) error {
 				)
 			}
 
-			if _, ok := seen[path.Prefix()]; ok {
-				return newUpdateError(
-					commonerrors.ErrConflictingUpdateOperators,
-					fmt.Sprintf(
-						"Updating the path '%[1]s' would create a conflict at '%[1]s'", key,
-					),
-					command,
-				)
-			}
+			for _, oldPath := range paths {
+				oldKeys := oldPath.Slice()
+				nextKeys := nextPath.Slice()
 
-			seen[path.Prefix()] = struct{}{}
+				var isPrefix bool
+
+				if len(oldKeys) > len(nextKeys) {
+					isPrefix = checkSlicePrefix(oldKeys, nextKeys)
+				} else {
+					isPrefix = checkSlicePrefix(nextKeys, oldKeys)
+				}
+
+				if isPrefix {
+					return newUpdateError(
+						commonerrors.ErrConflictingUpdateOperators,
+						fmt.Sprintf(
+							"Updating the path '%[1]s' would create a conflict at '%[1]s'", key,
+						),
+						command,
+					)
+				}
+			}
+			paths = append(paths, &nextPath)
 		}
 	}
 
 	return nil
 }
 
+// checkSlicePrefix returns true if prefix slice is the beginning of the target slice.
+// The example of slice prefix: target = ["a","b","c"] prefix = ["a","b"];
+// For empty prefix slice it returns false, if target slice is not empty.
+func checkSlicePrefix[T comparable](target, prefix []T) bool {
```

Comment:
Why do we need generics here?

Reply:
As it was easy to make this function more general I did it, but if we prefer strings here I can change that